### PR TITLE
Reduce some allocations

### PR DESCRIPTION
**Pull request by cvium**
**Changes**
Reduce allocations by using spans and by reducing the number of times a column value is retrieved (effectively halved).

### CODE REVIEW

File: Emby.Server.Implementations/HttpServer/Security/AuthorizationContext.cs [hunk 1]
```diff
@@ -232,51 +232,51 @@ private AuthorizationInfo GetAuthorization(HttpContext httpReq)
...
                 return null;
             }
 
-            var acceptedNames = new[] { "MediaBrowser", "Emby" };
+            var name = authorizationHeader[..firstSpace];
 
-            // It has to be a digest request
-            if (!acceptedNames.Contains(parts[0], StringComparer.OrdinalIgnoreCase))
+            if (!name.Equals("MediaBrowser", StringComparison.OrdinalIgnoreCase)
+                && name.Equals("Emby", StringComparison.OrdinalIgnoreCase))
```
Comment (crobibero) [inline Emby.Server.Implementations/HttpServer/Security/AuthorizationContext.cs:261]:
```suggestion
                && !name.Equals("Emby", StringComparison.OrdinalIgnoreCase))
```

Missed conditional?

Comment (cvium) [reply Emby.Server.Implementations/HttpServer/Security/AuthorizationContext.cs:261]: True

**Review by crobibero — APPROVED** [review]

File: tests/Jellyfin.Providers.Tests/MediaInfo/SubtitleResolverTests.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    CreateMediaStream("/video/My.Video.default.forced.en.srt", "srt", "en", index++, isForced: true, isDefault: true),
+                    CreateMediaStream("/video/My.Video.en.default.forced.srt", "srt", "en", index++, isForced: true, isDefault: true),
+                    CreateMediaStream("/video/My.Video.With.Additional.Garbage.en.srt", "srt", "en", index),
+                }
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(AddExternalSubtitleStreams_GivenMixedFilenames_ReturnsValidSubtitles_TestData))]
+        public void AddExternalSubtitleStreams_GivenMixedFilenames_ReturnsValidSubtitles(List<MediaStream> streams, string videoPath, int startIndex, string[] files, List<MediaStream> expectedResult)
```
Comment (Bond-009) [inline tests/Jellyfin.Providers.Tests/MediaInfo/SubtitleResolverTests.cs:61]: Doesn't an array make more sense for `expectedResult`?

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.Server.Implementations/Data/SqliteExtensions.cs [hunk 3]
```diff
@@ -96,46 +97,135 @@ public static DateTime ReadDateTime(this IResultSetValue result)
...
+            var item = reader[index];
+            if (item.IsDbNull())
+            {
+                return false;
             }
 
-            return null;
+            result = item.ReadGuidFromBlob();
+            return true;
         }
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteExtensions.cs:131]: These also shouldn't be nullable

File: Emby.Server.Implementations/Data/SqliteExtensions.cs [hunk 4]
```diff
@@ -96,46 +97,135 @@ public static DateTime ReadDateTime(this IResultSetValue result)
...
-            return result[index].ToFloat();
+            var item = reader[index];
+            if (item.IsDbNull())
+            {
+                result = default;
+                return false;
+            }
+
+            result = item.ToDouble();
+            return true;
```
Comment (Bond-009) [inline Emby.Server.Implementations/Data/SqliteExtensions.cs:228]: Remove the `[NotNullWhen(true)]`

**Review by Bond-009 — CHANGES_REQUESTED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**